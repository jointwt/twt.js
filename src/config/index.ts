import { APIConfig } from '../types.d';

export default {
  prefix: 'api/v1',
  api: {
    ping: {
      endpoint: 'ping',
      method: 'GET',
    },
    register: {
      endpoint: 'register',
      method: 'POST',
    },
    auth: {
      endpoint: 'auth',
      method: 'POST',
    },
    post: {
      endpoint: 'post',
      method: 'POST',
    },
    timeline: {
      endpoint: 'timeline',
      method: 'POST',
    },
    discover: {
      endpoint: 'discover',
      method: 'POST',
    },
    follow: {
      endpoint: 'follow',
      method: 'POST',
    },
  },
} as APIConfig;
