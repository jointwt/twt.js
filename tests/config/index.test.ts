import config from '../../src/config';

test('config should be defined', () => {
  expect(config).toMatchSnapshot();
});
