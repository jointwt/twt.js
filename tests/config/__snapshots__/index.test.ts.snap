// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config should be defined 1`] = `
Object {
  "api": Object {
    "auth": Object {
      "endpoint": "auth",
      "method": "POST",
    },
    "discover": Object {
      "endpoint": "discover",
      "method": "POST",
    },
    "follow": Object {
      "endpoint": "follow",
      "method": "POST",
    },
    "ping": Object {
      "endpoint": "ping",
      "method": "GET",
    },
    "post": Object {
      "endpoint": "post",
      "method": "POST",
    },
    "register": Object {
      "endpoint": "register",
      "method": "POST",
    },
    "timeline": Object {
      "endpoint": "timeline",
      "method": "POST",
    },
  },
  "prefix": "api/v1",
}
`;
